### PR TITLE
Cody completions: Various improvements to reliability and logging

### LINKED ISSUE #52686: Cody completions: Can we fix Thorsten's example of gibberish responses?

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #52686

There are a bunch of small changes in this PR. Here's a summary:

- I changed the default prompt to (hopefully) reduce the number of non-code responses. This works but mostly by gut feel. We'll need @valerybugakov's work to make sure we make some actual progress there. This was mostly used to reduce the number of times the completion starts with a code block end (```) tag so it increases the requests that make it through to the user.
- Filter out some known "bad starters" that we have collected over the last week. E.g. where cody starts the response with `➕     console.log('hi')` for example. 🙃 This means that more requests should make it to the user. 
- Instead of throwing away responses that start with whitespace, we now trim it instead. This should mean that more requests make it through the user.
- Empty responses (those that we filter out in the post process logic) are no longer forwarded to VS Code. This mainly fixes a small annoyance where the completions widget would e.g. show `1/2` completions but the second one being empty. 
- Tweaks based on [this research](https://sourcegraph.slack.com/archives/C05A7PPR07Q/p1685715846764189)
  - We now make 3 requests instead of 2 on empty lines
  - Reduced the debouncing timeout quite significantly so that our latency becomes a lot besser. 
  - ➡ We're working simultaneously to increase the rate-limit for dotcom
- Logging:
  - We now log whenever we have no proper completion to show (so we can assess how rare this is since it happened often in internal tests)
  - We now count completions as `read` when they are accepted (so that we do never count a completion as accepted but not as suggested)
- Removed some debug logging leftover 

## Test plan

Tests still green.

### CODE REVIEW

File: client/cody/CHANGELOG.md [hunk 1]
```diff
@@ -15,17 +15,20 @@ Starting from `0.2.0`, Cody is using `major.EVEN_NUMBER.patch` for release versi
 ### Fixed
 
 - Cody Inline Assist: Decorations for `/fix` on light theme [pull/52796](https://github.com/sourcegraph/sourcegraph/pull/52796)
-- Cody Inline Assist: Fixs cody processing indefinitely isse [pull/52796](https://github.com/sourcegraph/sourcegraph/pull/52796)
+- Cody Inline Assist: Fixes cody processing indefinitely issue [pull/52796](https://github.com/sourcegraph/sourcegraph/pull/52796)
```
Comment (abeatrix) [inline client/cody/CHANGELOG.md:18]: 🙏 thank you

**Review by abeatrix — APPROVED** [review]
Lgtm. 

On an unrelated note, I've encountered an issue before with suggestions returns code that starts with '''tag, maybe something that can be removed manually for completions? (we do that for code returns for fixup too).

File: client/cody/src/completions/multiline.ts [hunk 2]
```diff
@@ -47,9 +47,7 @@ export function truncateMultilineCompletion(
 
     // Normalize responses that start with a newline followed by the exact indentation of
     // the first line.
-    console.log(lines)
     if (lines.length > 1 && lines[0] === '' && indentation(lines[1]) === startIndent) {
-        console.log('WHAT')
```
Comment (valerybugakov) [inline client/cody/src/completions/multiline.ts:52]: Haha, when I saw these in the devtools, I was like: WHAT? I didn't add such logs yet.

**Review by valerybugakov — APPROVED** [review]
Nice, thanks for the detailed PR description!

**Comment by valerybugakov** [comment]
> would e.g. show 1/2 completions but the second one being empty.

Love it! Bumped into this behavior multiple times yesterday.

File: client/cody/src/completions/multiline.ts (hunk 2 above)
Comment (philipp-spiess) [reply client/cody/src/completions/multiline.ts:52]: 🙈

**Pull request merged**